code/math/spline.cpp

```diff
@@ -26,18 +26,28 @@
 // SPLINE FUNCTIONS
 //
 
-// integer factorial. =o
-int bez_fact(int n)
+static float bez_fact_lookup[13] = {
+	1.0f,          // 0!
+	1.0f,          // 1!
+	2.0f,          // 2!
+	6.0f,          // 3!
+	24.0f,         // 4!
+	120.0f,        // 5!
+	720.0f,        // 6!
+	5040.0f,       // 7!
+	40320.0f,      // 8!
+	362880.0f,     // 9!
+	3628800.0f,    // 10!
+	39916800.0f,   // 11!
+	479001600.0f,  // 12!
+};
+
+// Limited Factorial
+static float bez_fact(int n)
 {
-	int idx;
-	int product = 1;
-
-	// do eet
-	for(idx=1; idx<=n; idx++){
-		product *= idx;
-	}
+	Assert((n >= 0) && (n <= 12));
 
-	return product;
+	return bez_fact_lookup[n];
```

Comment:
Perhaps this is clear elsewhere, but what happens in non-debug runtime if `bez_fact()` is called outside of [0,12]? I'm partly answering my question when I say there will be an index access outside the bez_fact_lookup[] array.

Can we prove there are no indirect calls to `bez_fact()` that call outside [0,12]?

A doxygen comment on the function that notes the restriction would also be useful.